### PR TITLE
feat: implement onboarding for kubectl

### LINKED ISSUE #4118: Provide a way to install `kubectl` CLI
### Is your enhancement related to a problem? Please describe

We're providing a lot of CLI like compose, kind, etc

but we should also provide a way to setup `kubectl` binary

### Describe the solution you'd like

Have a nice way to install `kubectl`

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #4118
Relates to #5035

### What does this PR do?

Add onboarding for kubectl-cli. Code is mainly cloned from compose except for the download part which is not based on GitHub.

### What issues does this PR fix or reference?

References #5035 

### How to test this PR?

1. Remove kubectl from your PATH
2. Go to Settings -> Resources
3. Onboard kubectl

### CODE REVIEW

File: extensions/kubectl-cli/package.json [hunk 1]
```diff
@@ -4,12 +4,138 @@
...
+        "title": "kubectl: Checks..."
+      },
+      {
+        "command": "kubectl.onboarding.checkDownloadedCommand",
+        "title": "kubectl: Check kubectl Downloaded"
+      },
+      {
+        "command": "kubectl.onboarding.downloadCommand",
+        "title": "kubectl: Download kubectl"
+      }
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:30]: This is something that is not done neither in podman nor in compose but... should we hide these commands in the command palette? I don't see any reason to show them. 
We just need to add `enablement: false` to the commands we do not want to show https://github.com/containers/podman-desktop/pull/4630

File: extensions/kubectl-cli/package.json [hunk 2]
```diff
@@ -4,12 +4,138 @@
...
+        },
+        {
+          "id": "welcomeDownloadView",
+          "label": "kubectl Download",
+          "title": "kubectl Download",
+          "when": "onboardingContext:kubectlIsNotDownloaded == true",
+          "content": [
+            [
+              {
+                "value": "kubectl support in Podman Desktop enables you to use kubectl CLI ."
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:53]:
```suggestion
                "value": "Kubectl support in Podman Desktop enables you to use Kubectl CLI ."
```

File: extensions/kubectl-cli/package.json [hunk 3]
```diff
@@ -4,12 +4,138 @@
...
+          "title": "Checking for kubectl installation",
+          "command": "kubectl.onboarding.checkDownloadedCommand",
+          "completionEvents": [
+            "onCommand:kubectl.onboarding.checkDownloadedCommand"
+          ]
+        },
+        {
+          "id": "welcomeDownloadView",
+          "label": "kubectl Download",
+          "title": "kubectl Download",
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:48]:
```suggestion
          "title": "Kubectl Download",
```

File: extensions/kubectl-cli/package.json [hunk 4]
```diff
@@ -4,12 +4,138 @@
...
+          "when": "onboardingContext:kubectlIsNotDownloaded == true",
+          "content": [
+            [
+              {
+                "value": "kubectl support in Podman Desktop enables you to use kubectl CLI ."
+              }
+            ],
+            [
+              {
+                "value": "kubectl will be downloaded in the next step (Version ${onboardingContext:kubectlDownloadVersion}). :link[Want to download a different version?]{command=kubectl.onboarding.promptUserForVersion}",
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:58]:
```suggestion
                "value": "Kubectl will be downloaded in the next step (Version ${onboardingContext:kubectlDownloadVersion}). :link[Want to download a different version?]{command=kubectl.onboarding.promptUserForVersion}",
```

File: extensions/kubectl-cli/package.json [hunk 5]
```diff
@@ -4,12 +4,138 @@
...
+                "when": "!onboardingContext:kubectlShowCustomDownloadDialog"
+              }
+            ]
+          ]
+        },
+        {
+          "id": "downloadCommand",
+          "title": "Downloading kubectl ${onboardingContext:kubectlDownloadVersion}",
+          "description": "Downloading the binary.\n\nOnce downloaded, the next step will install kubectl system-wide.",
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:67]:
```suggestion
          "description": "Downloading the binary.\n\nOnce downloaded, the next step will install Kubectl system-wide.",
```

File: extensions/kubectl-cli/package.json [hunk 6]
```diff
@@ -4,12 +4,138 @@
...
+          "when": "onboardingContext:kubectlIsNotDownloaded == true",
+          "command": "kubectl.onboarding.downloadCommand",
+          "completionEvents": [
+            "onCommand:kubectl.onboarding.downloadCommand"
+          ]
+        },
+        {
+          "id": "downloadFailure",
+          "title": "Failed Downloading kubectl",
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:76]:
```suggestion
          "title": "Failed Downloading Kubectl",
```

File: extensions/kubectl-cli/package.json [hunk 7]
```diff
@@ -4,12 +4,138 @@
...
+          "when": "onboardingContext:kubectlIsNotDownloaded == true",
+          "state": "failed"
+        },
+        {
+          "id": "downloadSuccessfulView",
+          "title": "kubectl Successfully Downloaded",
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:82]:
```suggestion
          "title": "Kubectl Successfully Downloaded",
```

File: extensions/kubectl-cli/package.json [hunk 8]
```diff
@@ -4,12 +4,138 @@
...
+          "when": "onboardingContext:kubectlIsNotDownloaded == false",
+          "content": [
+            [
+              {
+                "value": "kubectl has been successfully downloaded! In order to use kubectl from a shell, it is required for kubectl to be installed system-wide.\n\nThe next step will install kubectl system-wide. **You will be prompted for system privileges when enabling this.**"
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:87]:
```suggestion
                "value": "Kubectl has been successfully downloaded! In order to use Kubectl from a shell, it is required for Kubectl to be installed system-wide.\n\nThe next step will install Kubectl system-wide. **You will be prompted for system privileges when enabling this.**"
```

File: extensions/kubectl-cli/package.json [hunk 9]
```diff
@@ -4,12 +4,138 @@
...
+              }
+            ]
+          ]
+        },
+        {
+          "id": "installSystemWideCommand",
+          "title": "Install kubectl",
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:94]:
```suggestion
          "title": "Install Kubectl",
```

File: extensions/kubectl-cli/package.json [hunk 10]
```diff
@@ -4,12 +4,138 @@
...
+              {
+                "value": "kubectl will be downloaded in the next step (Version ${onboardingContext:kubectlDownloadVersion}). :link[Want to download a different version?]{command=kubectl.onboarding.promptUserForVersion}",
+                "when": "!onboardingContext:kubectlShowCustomDownloadDialog"
+              }
+            ]
+          ]
+        },
+        {
+          "id": "downloadCommand",
+          "title": "Downloading kubectl ${onboardingContext:kubectlDownloadVersion}",
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:66]:
```suggestion
          "title": "Downloading Kubectl ${onboardingContext:kubectlDownloadVersion}",
```

File: extensions/kubectl-cli/package.json [hunk 11]
```diff
@@ -4,12 +4,138 @@
...
+          "description": "Installing the binary system-wide.\n\n You may be prompted for elevated system privileges.",
+          "when": "kubectl.isKubectlInstalledSystemWide == false",
+          "command": "kubectl.onboarding.installSystemWideCommand",
+          "completionEvents": [
+            "onCommand:kubectl.onboarding.installSystemWideCommand"
+          ]
+        },
+        {
+          "id": "installSystemWideFailure",
+          "title": "Failed Installing kubectl",
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:104]:
```suggestion
          "title": "Failed Installing Kubectl",
```

File: extensions/kubectl-cli/package.json [hunk 12]
```diff
@@ -4,12 +4,138 @@
...
+          "when": "kubectl.isKubectlInstalledSystemWide == false",
+          "state": "failed"
+        },
+        {
+          "id": "installSystemWideSuccess",
+          "title": "kubectl Successfully Installed",
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:110]:
```suggestion
          "title": "Kubectl Successfully Installed",
```

File: extensions/kubectl-cli/package.json [hunk 13]
```diff
@@ -4,12 +4,138 @@
...
+          "when": "kubectl.isKubectlInstalledSystemWide == true",
+          "state": "completed",
+          "content": [
+            [
+              {
+                "value": "kubectl has been successfully installed system-wide!"
```
Comment (lstocchi) [inline extensions/kubectl-cli/package.json:116]:
```suggestion
                "value": "Kubectl has been successfully installed system-wide!"
```

**Review by lstocchi — COMMENTED** [review]
Tested and works fine for me. Just some nits. 
For me the name Kubectl needs to start with a capital letter or atleast when it's the first word in a sentence

File: extensions/kubectl-cli/package.json (hunk 1 above)
Comment (benoitf) [reply extensions/kubectl-cli/package.json:30]: yes probably should be added

**Review by benoitf — APPROVED** [review]
I was able to install kubectl on macOS

if you can disable some commands from the command palette (or make them work if not part of onboarding (for example download or install could work like if no param, it prompts the user)


I noticed also that when I select the version, I have this prompt
image

it's sorted by release date but maybe as a user I would want to be sorted by version number like 1.28.4 then 1.28.3 etc (or release date should appear like 7d ago, 1y ago) but can be follow up

**Pull request merged**